display_list/skia/dl_sk_canvas.cc

```diff
@@ -49,6 +45,22 @@ static SkPaint ToSk(const DlPaint& paint, bool force_stroke = false) {
     color_filter = invert_filter;
   }
   sk_paint.setColorFilter(color_filter);
+
+  auto color_source = paint.getColorSourcePtr();
+  if (color_source) {
+    // On the Impeller backend, we will only support dithering of *gradients*,
+    // and it will be enabled by default (without the option to disable it).
+    // Until Skia support is completely removed, we only want to respect the
+    // dither flag for gradients (otherwise it will also apply to, for example,
+    // images, which is not supported in Impeller).
+    //
+    // See https://github.com/flutter/flutter/issues/112498.
+    if (color_source->isGradient()) {
+      sk_paint.setDither(paint.isDither());
```

Comment:
> I thought the purpose of this fix was that dithering happens when gradients are involved 

Yeah that's the end goal, but as written here:

> // On the Impeller backend, we will only support dithering of *gradients*,
> // and it will be enabled by default (without the option to disable it).
> // Until Skia support is completely removed ...

... that's an aspirational state. The current behavior (see issue https://github.com/flutter/engine/pull/44705) is that users *are* allowed to disable dithering, so we don't want to _unconditionally_ enable or disable dithering. I'm open to making this more clear - and given the back and forth on the PR, have done so already.